x/poolmanager/msg_server.go

```diff
@@ -122,5 +121,23 @@ func (server msgServer) SplitRouteSwapExactAmountOut(goCtx context.Context, msg
 }
 
 func (server msgServer) SetDenomPairTakerFee(goCtx context.Context, msg *types.MsgSetDenomPairTakerFee) (*types.MsgSetDenomPairTakerFeeResponse, error) {
-	return nil, errors.New("not implemented")
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	for _, denomPair := range msg.DenomPairTakerFee {
+		err := server.keeper.SenderValidationSetDenomPairTakerFee(ctx, msg.Sender, denomPair.Denom0, denomPair.Denom1, denomPair.TakerFee)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Set denom pair taker fee event is handled in each iteration of the loop above
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.Sender),
+		),
+	})
+
+	return &types.MsgSetDenomPairTakerFeeResponse{Success: true}, nil
```

Comment:
Why do we have a success field in response, should we just remove this?

(not returning an error => success, right)

Reply:
Follows the same format as other messages with no important return values. If we decide this, we should remove it from all of them at a single time imo.

Reply:
I'm unaware of us doing this convention. I at least think we shouldn't introduce the design pattern more, unless this is replicated in the SDK